packages/client-sdk-web/src/internal/auth-client.ts

```diff
@@ -0,0 +1,73 @@
+import {auth} from '@gomomento/generated-types-webtext';
+import {MomentoLogger, GenerateApiToken} from '..';
+import {version} from '../../package.json';
+import {Configuration} from '../config/configuration';
+import {Request, UnaryInterceptor, UnaryResponse} from 'grpc-web';
+import {Header, HeaderInterceptorProvider} from './grpc/headers-interceptor';
+import {_GenerateApiTokenRequest} from '@gomomento/generated-types-webtext/dist/auth_pb';
+import {cacheServiceErrorMapper} from '../errors/cache-service-error-mapper';
+import {_GenerateApiTokenResponse} from '@gomomento/common';
+import Never = _GenerateApiTokenRequest.Never;
+import Expires = _GenerateApiTokenRequest.Expires;
+
+export interface AuthClientProps {
```

Comment:
This is a public interface, which users will have to instantiate in order to create the client.  So it shouldn't live in an `internal` file or else they would have to import it from there.  We will either need to move it to the main auth-client.ts file, or put it in its own file to avoid circular dependencies between the two files.

Also - you currently have the `controlEndpoint` specified as a required parameter.  In the 90% use case if a customer instantiates this client we don't want them to have to know that endpoint, and we should be able to use a default one, right?

Neither of these have to be addressed in this PR as long as we are not touching the nodejs SDK (i.e. remove the file I commented on above), but we should create tickets for them if we aren't going to address them now.  I do want to get this merged to unblock you to finish the console work.

Reply:
Removed `AuthClient` from `client-sdk-nodejs` and created the tickets to address your feedback:
- https://github.com/momentohq/client-sdk-nodejs/issues/395
- https://github.com/momentohq/client-sdk-nodejs/issues/396